### PR TITLE
fix(c++): fix the 'multiple definition' bug when linking libarrow.a and libgraphar_bundled_dependencies.a in one CMakeLists.txt

### PR DESCRIPTION
**Pull request by Elssky**
fix the issues in https://github.com/apache/incubator-graphar/pull/628

### CODE REVIEW

**Comment by Elssky** [comment]
Hi,@acezen @jasinliu.  This should solve the problem in https://github.com/apache/incubator-graphar/pull/628, please review this.

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -231,21 +319,40 @@ macro(build_graphar_with_arrow_bundled)
...
+                graphar_bundled_dependencies
                 "-framework CoreFoundation"
                 "-framework Security")
     else()
-        target_link_libraries(graphar PRIVATE -Wl,--exclude-libs,ALL -Wl,--whole-archive gar_arrow_static
-                "${GAR_PARQUET_STATIC_LIB}"
-                "${GAR_DATASET_STATIC_LIB}"
-                "${GAR_ARROW_ACERO_STATIC_LIB}"
-                "${GAR_ARROW_BUNDLED_DEPS_STATIC_LIB}" -Wl,--no-whole-archive) 
+        target_link_libraries(graphar PRIVATE -Wl,--exclude-libs,ALL
```
Comment (Elssky) [inline cpp/CMakeLists.txt:354]: remove `-Wl,--whole-archive` and  `-Wl,--no-whole-archive`

**Comment by jasinliu** [comment]
> Hi,@acezen @jasinliu. This should solve the problem in #628, please review this.

Thanks! I think you can create some tests following #628 . Or you can provide the verification process through comments.

**Comment by Elssky** [comment]
> > Hi,@acezen @jasinliu. This should solve the problem in #628, please review this.
> 
> Thanks! I think you can create some tests following #628 . Or you can provide the verification process through comments.

Yes, I used the totally same exmaple in the comment for testing and the installation steps are exactly the same, the results show that we can run this code correctly
image
## Other example 
### test_read.cc
``` cpp
#include <cstdlib>
#include <iostream>
#include "arrow/api.h"
#include "graphar/api/arrow_reader.h"



int main() {
    std::string path = "/workspaces/incubator-graphar/testing/ldbc_sample/parquet/ldbc_sample.graph.yml";
    std::string src_type = "person", edge_type = "knows", dst_type = "person";
    std::string vertex_property_name = "id";
    std::string edge_property_name = "creationDate";
    auto maybe_graph_info = graphar::GraphInfo::Load(path);
    auto graph_info = maybe_graph_info.value();
    auto vertex_info = graph_info->GetVertexInfo(src_type);


    auto maybe_reader = graphar::VertexPropertyArrowChunkReader::Make(
    graph_info, src_type, vertex_property_name);
    auto reader = maybe_reader.value();
    std::cout << "Chunknum " << reader->GetChunkNum() << std::endl;

    auto result = reader->GetChunk();
    auto table = result.value();
    std::cout << "num_rows " << table->num_rows() << std::endl;

}
```
### CMakeLists.txt
```cmake
cmake_minimum_required(VERSION 3.15)

project(test_read)

set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -Wall")

set(TARGET test_read)

# find_package(OpenSSL REQUIRED)
find_package(graphar REQUIRED)
find_package(Arrow REQUIRED)

add_executable(${TARGET} test_read.cc)
target_link_libraries(${TARGET} PRIVATE graphar Arrow::arrow_static)
```
### build
```shell 
cmake . && make
```
image

**Review by acezen — APPROVED** [review]
LGTM

**Pull request merged**